src/mono/msbuild/android/build/AndroidBuild.targets

```diff
@@ -38,10 +37,11 @@
       <AndroidBundleDir Condition="'$(AndroidBundleDir)' == ''">$([MSBuild]::NormalizeDirectory('$(OutDir)', 'Bundle'))</AndroidBundleDir>
 
       <BundleDir>$(AndroidBundleDir)</BundleDir>
-      <_MonoHeaderPath>$([MSBuild]::NormalizeDirectory($(MicrosoftNetCoreAppRuntimePackRidNativeDir), 'include', 'mono-2.0'))</_MonoHeaderPath>
+      <RuntimeHeaders>$([MSBuild]::NormalizeDirectory($(MicrosoftNetCoreAppRuntimePackRidNativeDir), 'include', 'mono-2.0'))</RuntimeHeaders>
```

Comment:
Why do we need to include `mono-2.0` in the header for CoreCLR? On Apple it is conditioned by `'$(UseMonoRuntime)' != 'false'...`

Reply:
This is the default behavior, the line bellow will change the `RuntimeHeaders` path if we target CoreCLR, and we should do that on Apple as well.

Apple currently has different mechanism because Filip's PR (coreCLR on iossim bring up) was kind of done at the same time so he did it through: https://github.com/dotnet/runtime/blob/bfa02766bdde4657ddd1f734e274d43e6ba3c036/src/tasks/AppleAppBuilder/AppleAppBuilder.csproj#L12 

That is something we would clean up in:  https://github.com/dotnet/runtime/issues/111923